python/tests/test_high_level.py

```diff
@@ -221,3 +221,24 @@ def from_numeric(n: Numeric) -> OtherNumeric:  # type: ignore[empty-body]
 
     egraph.register(rewrite(OtherNumeric(1)).to(from_numeric(Numeric.ONE)))
     assert expr_parts(egraph.simplify(OtherNumeric(i64(1)), 10)) == expr_parts(from_numeric(Numeric.ONE))
+
+
+def test_f64_negation() -> None:
+    egraph = EGraph()
+    # expr1 = -2.0
+    expr1 = egraph.define("expr1", -f64(2.0))
+
+    # expr2 = 2.0
+    expr2 = egraph.define("expr2", f64(2.0))
+
+    # expr3 = -(-2.0)
+    expr3 = egraph.define("expr3", -(-f64(2.0)))
+
+    x, y = vars_("x y", f64)
+
+    egraph.register(rewrite(-(-x)).to(x))
+
+    egraph.run(10)
```

Comment:
Could you try the test without these lines?

Since the `neg` is defined as a primitive, it shouldn't need any re-rewrites to execute.

Reply:
Ah, thanks for the pointer! Sorry, I'm quite new to e-graphs...

Reply:
No problem... There is also the added complexity of understanding what is a primitive operation in egglog and what is not, which I still sometimes find un-intuitive.